docs/corfu-queue-design.md

```diff
@@ -0,0 +1,12 @@
+The class `CorfuQueue` implements a persisted queue over the abstraction of a CorfuTable. Currently CorfuTable uses a HashMap<> to represent the materialized view of a distributed map. A HashMap does not carry a notion of ordering natively. However when implemented over the abstraction of a distributed shared log, the elements added to the map do, in fact, have an ordering imposed by their append or transaction commit operations into the global log. The class `CorfuQueue` attempts to expose this inherent ordering as a persisted Queue with three simple apis:
+
+###  1. `UUID enqueue(E)`
+Since a map has a key and value, where key is a conflict parameter, enqueue generates a non-conflicting Long as the key and inserts the Entry as a value into a CorfuTable. The generated Long is packed into the LSB of a UUID and returned to the caller.
```

Comment:
Should we also note the fact that the UUID returned by enqueue is different from the one returned by entryList? Since two different UUIDs can reference the same entry it seems like a weird semantic that we should be clear about (i.e. these are "special uuids")

Reply:
The api does note it clearly, but let me add it to the design doc as well.